src/backend/app.js

```diff
@@ -21,10 +22,86 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+//Respond with all meals in the future (relative to the when datetime) - /future-meals
+
+app.get("/future-meals", async (req, res) => {
+  try {
+    const today = new Date();
+    const futureMeals = await knex("meal").where("event_date", ">", today);
+    if (futureMeals.length === 0) {
+      res.status(204).json(futureMeals);
+    }
+    res.status(200).json(futureMeals);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Server Error");
+  }
+});
+
+//Respond with all meals in the past (relative to the when datetime) - /past-meals
+
+app.get("/past-meals", async (req, res) => {
+  try {
+    const today = new Date();
+    const pastMeals = await knex("meal").where("event_date", "<", today);
+    if (pastMeals.length === 0) {
+      res.status(204).json(pastMeals);
+    }
+    res.status(200).json(pastMeals);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Server Error");
+  }
+});
+
+//Respond with all meals sorted by ID - /all-meals
+
+app.get("/all-meals", async (req, res) => {
+  try {
+    const allMeals = await knex("meal").orderBy("id");
+    res.status(200).json(allMeals);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Server Error");
+  }
+});
+
+//Respond with the first meal (meaning with the minimum id) - /first-meal
+
+app.get("/first-meal", async (req, res) => {
+  try {
+    const firstMeal = await knex("meal").orderBy("id").first();
+    if (firstMeal !== 0) {
+      res.status(200).json(firstMeal);
+    } else {
+      res.status(404).send("There is no first meal");
+    }
```

Comment:
it's better to handle first the empty case, which means if we don't have firstMeal and then we send the the response with meal 
you can do 
`if(!firstMeal){
res.status(204).json("There is no first meal");
}
res.status(200).json(firstMeal;`